### PR TITLE
Improve monogo and snowflake detectors

### PR DESCRIPTION
**Pull request by dustin-decker**
### Description:
- uses lighter snowflake driver
- adds timeouts to mongo detector

### Checklist:
* [X] Tests passing (`make test-community`)?
* [X] Lint passing (`make lint` this requires [golangci-lint](https://golangci-lint.run/usage/install/#local-installation))?

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -4,6 +4,8 @@ go 1.21
 
 replace github.com/jpillora/overseer => github.com/trufflesecurity/overseer v1.2.7
 
+replace github.com/snowflakedb/gosnowflake => github.com/trufflesecurity/gosnowflake v0.0.1
```
Comment (mcastorina) [inline go.mod:7]: I see [trufflesecurity/gosnowflake](https://github.com/trufflesecurity/gosnowflake) is forked from `snowflakedb/gosnowflake` with a lot of code deleted. What is the affect of doing that?

Also, how much effort do you anticipate it'll be for us to maintain this fork?

File: pkg/detectors/mongodb/mongodb.go [hunk 2]
```diff
@@ -85,6 +87,30 @@ func isErrDeterminate(err error) bool {
...
+			switch k {
+			case "tls":
+				if v[0] == "false" {
+					params.Set("tls", "false")
+				} else {
+					params.Set("tls", "true")
+				}
+			}
+		}
+	}
```
Comment (mcastorina) [inline pkg/detectors/mongodb/mongodb.go:107]: **Suggestion:**

```go
if v := parsed.Query().Get("tls"); v == "false" {
    params.Set("tls", "false")
} else if v != "" {
    params.Set("tls", "true")
}
```

File: pkg/detectors/mongodb/mongodb.go [hunk 3]
```diff
@@ -85,6 +87,30 @@ func isErrDeterminate(err error) bool {
...
+	params.Set("connectTimeoutMS", "3000")
+	params.Set("timeoutMS", "3000")
```
Comment (mcastorina) [inline pkg/detectors/mongodb/mongodb.go:109]: Should this be using the `timeout` argument?

**Review by mcastorina — APPROVED** [review]

File: pkg/detectors/mongodb/mongodb.go [hunk 4]
```diff
@@ -85,6 +87,30 @@ func isErrDeterminate(err error) bool {
 }
 
 func verifyUri(uri string, timeout time.Duration) error {
+	parsed, err := url.Parse(uri)
+	if err != nil {
+		return err
+	}
+
+	params := url.Values{}
+	for k, v := range parsed.Query() {
```
Comment (rgmz) [inline pkg/detectors/mongodb/mongodb.go:96]: The current MongoDB pattern needs to be updated to handle query parameters  (see #1550).

File: pkg/detectors/mongodb/mongodb.go (hunk 2 above)
Comment (dustin-decker) [reply pkg/detectors/mongodb/mongodb.go:107]: I added this in a way to easily add more parameters in the future, even if it's not the most efficient.

File: pkg/detectors/mongodb/mongodb.go (hunk 4 above)
Comment (dustin-decker) [reply pkg/detectors/mongodb/mongodb.go:96]: Oh, I'll take a look. Thank you.

**Pull request merged**